cmd/clusterctl/test/e2e/clusterctl.go

```diff
@@ -0,0 +1,159 @@
+		strings.Join(input.controlPlaneProviders, ", "),
+		strings.Join(input.infrastructureProviders, ", "),
+	))
+
+	//clusterctllog.SetLogger(newWriterLogger(input.writer))
+	c, err := clusterctlclient.New(clusterctlConfig.Path)
+	if err != nil {
+		return errors.Wrap(err, "failed to create the clusterctl client library")
+	}
+
+	initOpt := clusterctlclient.InitOptions{
+		Kubeconfig:              input.managementCluster.KubeconfigPath(),
+		CoreProvider:            input.coreProvider,
+		BootstrapProviders:      input.bootstrapProviders,
+		ControlPlaneProviders:   input.controlPlaneProviders,
+		InfrastructureProviders: input.infrastructureProviders,
+	}
+
+	providers, err := c.Init(initOpt)
+	if err != nil {
+		return errors.Wrap(err, "failed to run clusterctl init")
+	}
+
+	By("Waiting for installed providers")
+	for _, provider := range providers {
+		if err := input.managementCluster.WaitForProvider(ctx, provider.Name(), provider.Type(), ""); err != nil {
+			return errors.Wrapf(err, "error waiting for %s provider", provider.ManifestLabel())
+		}
+	}
+	return nil
+}
+
+// CreateWorkloadClusterInput is the input for CreateWorkloadCluster.
+type CreateWorkloadClusterInput struct {
+	clusterctlConfig         *ClusterctlConfig
+	managementCluster        *ManagementCluster
+	infrastructureProvider   string
+	name                     string
+	kubernetesVersion        string
+	controlPlaneMachineCount int
+	workerMachineCount       int
+}
+
+// CreateWorkloadCluster creates workload cluster.
+func CreateWorkloadCluster(ctx context.Context, input CreateWorkloadClusterInput) (*WorkloadCluster, error) {
+	// Creates a new workload cluster with name
+	By(fmt.Sprintf("Creating a workload cluster with name %s", input.name))
+
+	//clusterctllog.SetLogger(newWriterLogger(input.writer))
+	c, err := clusterctlclient.New(clusterctlConfig.Path)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create the clusterctl client library")
+	}
+
+	options := clusterctlclient.GetClusterTemplateOptions{
+		Kubeconfig: input.managementCluster.KubeconfigPath(),
+		ProviderRepositorySource: &clusterctlclient.ProviderRepositorySourceOptions{
+			InfrastructureProvider: input.infrastructureProvider,
+			Flavor:                 "",
+		},
+		ClusterName:              input.name,
+		KubernetesVersion:        input.kubernetesVersion,
+		ControlPlaneMachineCount: input.controlPlaneMachineCount,
+		WorkerMachineCount:       input.workerMachineCount,
+	}
+
+	template, err := c.GetClusterTemplate(options)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read workload cluster template")
+	}
+
+	yaml, err := template.Yaml()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate yaml for the workload cluster template")
+	}
+
+	if err := input.managementCluster.Apply(ctx, yaml); err != nil {
+		return nil, errors.Wrap(err, "failed to apply the workload cluster template")
+	}
+
+	return &WorkloadCluster{}, nil
```

Comment:
Is this just a place holder? Do you intend to put anything in the WorkloadCluster struct? I believe if the workload cluster is created w/o error, then all we need is the ClusterName to interact with it. In which case, we can just remove this?

Reply:
now it is addressed